### PR TITLE
jv - update list when done adding anime

### LINKED ISSUE #141: Update the userlist after adding an anime
After adding an anime form the Search modal to a userlist, refetch the userlist on the clientside to reflect new additions

Acceptance:
- [x] UserList UI refreshes with up to date information

### PR DESCRIPTION
**Pull request by jvoucsb**
## Notes
- Whenever the add anime modal closes we just refetch the userlist query 
- Whenever fecthing the new page we have to not include duplicates whenever concatting the list items
- **Need to update the MAX_PAGE as well** I don't think the MAX_PAGE will update properly for the case if we add anime that causes a new page to be made. So then the userlist wont go to the next page and display newly added anime. this can be done in separate issue later on.

## Testing
- Manual testing. The list updates correctly if we have multiple pages or one page

Closes #141
addlistrefresh

### CODE REVIEW

File: client/src/components/list/_UserList.tsx [hunk 1]
```diff
@@ -80,9 +86,12 @@ const UserList: React.FC<UserListProps> = ({ userlist, isOwn }) => {
         coverImage: anilistMedia.coverImage.medium,
         bannerImage: anilistMedia.bannerImage
       }));
-
+      
       setListItems(
-        prev => prev.concat(listItems)
+        prev => {
+          const ids = new Set<number>(prev.map(item => item.id));
+          return page === 1 ? listItems : prev.concat(listItems.filter(item => !ids.has(item.id)));
```
Comment (rowantran) [inline client/src/components/list/_UserList.tsx:93]: could you explain why we check if the page is 1 here?

**Review by rowantran — APPROVED** [review]
LGTM! not directly related to this PR but can we still add anime to user lists from the standalone search page or are we only able to do that from the "add anime" button on the list?

File: client/src/components/list/_UserList.tsx (hunk 1 above)
Comment (jvoucsb) [reply client/src/components/list/_UserList.tsx:93]: Oops I think it's leftover from previous nonworking implementation and it's not really needed

**Comment by jvoucsb** [comment]
> LGTM! not directly related to this PR but can we still add anime to user lists from the standalone search page or are we only able to do that from the "add anime" button on the list?

You are only able to add anime to a specific using the add button. Specifically by only this component: https://github.com/ucsb-cs148-f21/project-t10-animelist/blob/main/client/src/components/search/_SearchAddAnime.tsx

The old standalone page only works with the old one user list for every user implementation. I think it would be removed in migration #84  or have to refactored to account for multiple lists.

**Comment by jvoucsb** [comment]
I'm commenting on a corner case but its not really breaking.

**Need to update the MAX_PAGE as well** I don't think the MAX_PAGE will update properly for the case if we add anime that causes a new page to be made. So then the userlist wont go to the next page and display newly added anime. this can be done in separate issue later on.

**Review by christopherbarnett — APPROVED** [review]
cb - looks good to me!

**Pull request merged**